plugin/gcp/storageBucket.go

```diff
@@ -0,0 +1,70 @@
+package gcp
+
+import (
+	"context"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/puppetlabs/wash/activity"
+	"github.com/puppetlabs/wash/plugin"
+	"google.golang.org/api/iterator"
+)
+
+type storageBucket struct {
+	plugin.EntryBase
+	storageProjectClient
+}
+
+func newStorageBucket(client storageProjectClient, bucket *storage.BucketAttrs) *storageBucket {
+	stor := &storageBucket{EntryBase: plugin.NewEntry(bucket.Name), storageProjectClient: client}
+	stor.Attributes().SetMeta(bucket)
+	return stor
+}
+
+// List all storage objects as dirs and files.
+func (s *storageBucket) List(ctx context.Context) ([]plugin.Entry, error) {
+	bucket := s.Bucket(s.Name())
+	return listBucket(ctx, bucket, "")
+}
+
+func (s *storageBucket) Schema() *plugin.EntrySchema {
+	return plugin.NewEntrySchema(s, "bucket").SetMetaAttributeSchema(storage.BucketAttrs{})
+}
+
+func (s *storageBucket) ChildSchemas() []*plugin.EntrySchema {
+	return bucketSchemas()
+}
+
+const delimiter = "/"
+
+func listBucket(ctx context.Context, bucket *storage.BucketHandle, prefix string) ([]plugin.Entry, error) {
+	var entries []plugin.Entry
+	// Get objects directly under this prefix.
+	it := bucket.Objects(ctx, &storage.Query{Delimiter: delimiter, Prefix: prefix})
+	for {
+		objAttrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		if objAttrs.Prefix != "" {
```

Comment:
A high-level comment describing these checks (and how the storage API works) would be useful, because this isn't obvious to me w/o having to dig into the package myself.